### PR TITLE
Migrate magic numbers for status codes to named constants

### PR DESCRIPTION
**Pull request by amoghrajesh**
Building up on https://github.com/apache/airflow/pull/43663, replacing few of newer occurences of these magic number status codes to named constants as recommended by https://fastapi.tiangolo.com/reference/status/

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/api_fastapi/core_api/routes/public/assets.py [hunk 1]
```diff
@@ -80,7 +80,7 @@ def _generate_queued_event_where_clause(
 
 @assets_router.get(
     "/assets",
-    responses=create_openapi_http_exception_doc([401, 403, 404]),
+    responses=create_openapi_http_exception_doc([status.HTTP_404_NOT_FOUND]),
```
Comment (amoghrajesh) [inline airflow/api_fastapi/core_api/routes/public/assets.py]: Also removing the 401 and 403 as it comes from the top level router now

Comment (rawwar) [reply airflow/api_fastapi/core_api/routes/public/assets.py]: https://github.com/apache/airflow/pull/44161 already doing this

Comment (amoghrajesh) [reply airflow/api_fastapi/core_api/routes/public/assets.py]: Oh I see! Once your PR is merged, I will rebase and remove the duplicate ones

Comment (rawwar) [reply airflow/api_fastapi/core_api/routes/public/assets.py]: Its already approved. Can you merge it?

**Review by kaxil — APPROVED** [review]

**Pull request merged**